name: Run Unit Tests
on:
  workflow_dispatch:
  push:

jobs:
  unit-tests:
    uses: openvoiceos/.github/.github/workflows/pytest_file_or_dir.yml@feat/shared_actions1
    with:
      # runner:
      # branch:
      action_branch: "feat/shared_actions1"
      system_deps: jq
      python_matrix: '[3.8, 3.9, "3.10", "3.11", "3.12"]'
      pip_packages: pytest setuptools
      install_extras: "test,dev" # Don't actually exist but that should be ok
      timeout_minutes: 1 # Test multiple
      test_location: test
      is_skill: true
      codecov: true
      append_coverage: true
      upload_coverage: true
  unit-tests-longer-timeout:
    uses: openvoiceos/.github/.github/workflows/pytest_file_or_dir.yml@feat/shared_actions1
    with:
      # runner:
      # branch:
      action_branch: "feat/shared_actions1"
      system_deps: jq
      python_matrix: '["3.10"]'
      pip_packages: pytest
      install_extras: "test,dev" # Don't actually exist but that should be ok
      timeout_minutes: 15
      test_location: test
      is_skill: true
      codecov: true
      append_coverage: true
      upload_coverage: true
  unit-tests-longer-no-codecov:
    uses: openvoiceos/.github/.github/workflows/pytest_file_or_dir.yml@feat/shared_actions1
    with:
      # runner:
      # branch:
      action_branch: "feat/shared_actions1"
      system_deps: jq
      python_matrix: '["3.10"]'
      pip_packages: pytest
      install_extras: "test,dev" # Don't actually exist but that should be ok
      timeout_minutes: 15
      test_location: test
      is_skill: true # TODO: Should test false as well, but not on this repo
      codecov: false
